src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -304,8 +331,12 @@ internal static void WriteToStream(Stream input, Stream output, PSCmdlet cmdlet,
                     cmdlet.WriteProgress(record);
                 }
             }
-            catch (OperationCanceledException)
+            catch (OperationCanceledException exc)
             {
+                // We cannot distinct 'cancel` (user press Ctrl-C) and 'timeout' so we report the exception as-is
```

Comment:
Actually, we can differentiate them. Just check if `cancellationToken` is cancelled. If `Ctrl+c` is pressed, that token will be cancelled, otherwise, it's the timeout.

Reply:
Yes, you are right, we can (cancellationToken vs cts). Fixed. Also renamed `cts` to `timeoutCTS`.